dev_tools/scripts/create_new_algorithm.sh

```diff
@@ -172,44 +172,44 @@ TEMPLATE_FILES_DESCRIPTIONS=(
     "This is where tests with the DPC++ executor should be implemented. Usually, this means comparing against a Reference execution."
 )
 
-mkdir ${TMPDIR}
+mkdir "${TMPDIR}"
 
 for (( i=1; i<${#TEMPLATE_FILES[@]}+1; i++ ))
 do
-    sourcename=$(echo ${TEMPLATE_FILES[$i-1]} | sed "s/${name}/${source_name}/" )
+    sourcename=$(echo "${TEMPLATE_FILES[$i-1]}" | sed "s/${name}/${source_name}/" )
     sourcepath=${TEMPLATE_FILES_LOCATIONS[$i-1]}/${sourcename}
 
     # create folder for temporary files
-    mkdir -p ${TMPDIR}/${TEMPLATE_FILES_LOCATIONS[$i-1]}
+    mkdir -p "${TMPDIR}/${TEMPLATE_FILES_LOCATIONS[$i-1]}"
 
     # Evaluate the extension and try to find the matching files
-    for j in $(ls ${GINKGO_ROOT_DIR}/${sourcepath})
+    for j in "${GINKGO_ROOT_DIR}/${sourcepath}/*";
```

Comment:
The `ls` version is really required here, because we can have glob in the `sourcepath` variables (see the templates).
```suggestion
    for j in $(ls ${GINKGO_ROOT_DIR}/${sourcepath});
```

Reply:
But then we have to be really careful not to have any whitespace/IFS characters in there.

I'll check it out.

Reply:
Hmm, if you don't quote `${sourcepath}`, then it shouldn't be a problem.

```
for j in "${GINKGO_ROOT_DIR}/"${sourcepath}/*;
```
Produces a newline separated list of the files.

```
for j in "${GINKGO_ROOT_DIR}/${sourcepath}/"*;
```

Produces a space separated list of the files. Weird.